hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/metrics/TestS3GatewayMetrics.java

```diff
@@ -272,11 +273,367 @@ public void testPutAclFailure() throws Exception {
     assertEquals(1L, curMetric - oriMetric);
   }
 
+
```

Comment:
These tests are good but not sufficient. There are multiple failure paths possible and I do not think these test cases insures all the failure paths result in metrics being updated correctly. I would also add (or replace) to these tests but adding metrics check where the API is being tested. So when an API test expects a success/failure one of the asserts includes check for metric increments appropriately.

Reply:
We can file a follow up jira to add additional tests

Reply:
https://issues.apache.org/jira/browse/HDDS-6688